src/TimeTravel.elm

```diff
@@ -3,3 +3,104 @@ module TimeTravel exposing (addTimeTravel)
 import Playground exposing (..)
 import Set
 
+controlBarHeight = 64
+maxVisibleHistory = 2000
+
+addTimeTravel rawGame = 
+  { initialState = initialStateWithTimeTravel rawGame
+  , updateState = updateWithTimeTravel rawGame
+  , view = viewWithTimeTravel rawGame
+  }
+
+initialStateWithTimeTravel rawGame = 
+  { rawModel = rawGame.initialState
+  , paused = False
+  , history = []
+  , historyPlaybackPosition = 0
+  }
+
+updateWithTimeTravel rawGame computer currentModel =
+  let
+    updatedModel = 
+      { currentModel
+      | rawModel = rawGame.updateState computer currentModel.rawModel
+      , history = currentModel.history ++ [computer]
+      , historyPlaybackPosition = currentModel.historyPlaybackPosition + 1
+      }
+    newPlaybackPosition = 
+      min
+        (mousePosToHistoryIndex computer)
+        (List.length currentModel.history)
+    replayHistory pastInputs = 
+      List.foldl 
+        (\input model -> rawGame.updateState input model)
+        rawGame.initialState
+        pastInputs
+  in
+    if currentModel.paused && computer.mouse.down then
+      { currentModel
+      | historyPlaybackPosition = newPlaybackPosition
+      , rawModel = replayHistory (List.take newPlaybackPosition currentModel.history)
+      }
+    else if keyPressed "R" computer then
+      { updatedModel
+      | paused = False
+      , history = List.take currentModel.historyPlaybackPosition currentModel.history
+      }
+    else if keyPressed "T" computer then
+      { currentModel | paused = True }
+    else if keyPressed "C" computer then
+      { currentModel
+      | rawModel = rawGame.initialState
+      , history = []
+      , historyPlaybackPosition = 0
+      }
+    else if currentModel.paused then
+      currentModel
+    else
+      updatedModel
+
+viewWithTimeTravel rawGame computer model =
+  let
+    helpMessage =
+      if model.paused then
+        [ words white "Press R to resume or C to reset"
+          |> move 0 (computer.screen.top - controlBarHeight / 2 + 10)
```

Comment:
I wanted to have this split across two lines, while you had it on one. I think this is the simplest way to split across newlines (putting in \n just gave me a space), though I didn't look into it extensively.